File: core-arch/src/main/java/com/huyingbao/core/module/BaseModule.java
package com.huyingbao.core.module;

import android.app.Application;
import android.arch.lifecycle.ViewModelProvider;
import android.content.Context;

import com.huyingbao.core.store.RxStoreFactory;

import dagger.Binds;
import dagger.Module;

/**
 * Created by liujunfeng on 2017/12/7.
 */
@Module
public abstract class BaseModule {
    @Binds
    abstract Context privateContext(Application application);

    @Binds
    abstract ViewModelProvider.Factory privateViewModelFactory(RxStoreFactory rxStoreFactory);
}

File: core-base/src/main/java/com/huyingbao/core/base/BaseModule.kt
package com.huyingbao.core.base

import com.huyingbao.core.arch.module.RxFluxModule
import dagger.Module
import dagger.Provides
import dagger.android.AndroidInjectionModule
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import java.util.concurrent.TimeUnit
import javax.inject.Singleton

/**
 * 基础依赖注入仓库，所有子模块中的AppModule中需要包括该类，实现Store注入和Dagger.Android注入
 *
 * Created by liujunfeng on 2019/1/1.
 */
@Module(includes = [
    RxFluxModule::class,
    AndroidInjectionModule::class
])
class BaseModule {
    @Singleton
    @Provides
    fun provideClientBuilder(): OkHttpClient.Builder {
        //设置日志拦截器
        val interceptor = HttpLoggingInterceptor()
        interceptor.level = if (BuildConfig.DEBUG)
            HttpLoggingInterceptor.Level.BODY
        else
            HttpLoggingInterceptor.Level.NONE
        return OkHttpClient.Builder()
                .connectTimeout(BaseConstants.Config.HTTP_TIME_OUT, TimeUnit.SECONDS)
                .readTimeout(BaseConstants.Config.HTTP_TIME_OUT, TimeUnit.SECONDS)
                .writeTimeout(BaseConstants.Config.HTTP_TIME_OUT, TimeUnit.SECONDS)
                .addInterceptor(interceptor)
    }
}